bin/sum_product.py

```diff
@@ -103,9 +110,9 @@ def tensor_to_string(t):
 
     if args.trace:
         for el, zel in zs.items():
-            print(el.name, tensor_to_string(zel))
+            print(tensor_to_dict_string(args.pretty, fgg, el, zel))
```

Comment:
I wonder if the two remaining uses of `tensor_to_string` in this file can be changed to also use `tensor_to_dict_string` instead? (Then `tensor_to_string` should be inlined into `tensor_to_dict_string`.)

Reply:
This has been done in [the latest commit](https://github.com/chihyang/fggs/blob/780ec518d9acffb07d4a3275334461383a10b445/bin/sum_product.py#L123-L132).